apps/system/lockscreen/js/lockscreen.js

```diff
@@ -64,6 +64,15 @@
     */
     passCodeEnabled: false,
 
+	/*
+    * Boolean return whether pushing the power button locks screen
+    * immediately without waiting for lock timeout to pass.
+    * Must not multate directly - use setPassCodeEnabled(val)
+    * Only Settings Listener should change this value to sync with data
+    * in Settings API.
+    */
+    powerButtonLocksEnabled: false,
```

Comment:
I think it would be better to put your logic into a new LockScreen component under `system/lockscreen/js` and name it as `lockscreen_powerbutton_manager.js` to deal with all these. Although it seems to be a small component, to avoid growing this legacy LockScreen and put each functions in individual components is important. The new component, as a temporary workaround before we could really break all functions into widgets/components of screen lock, can still hold the LockScreen as it's one argument at starting up to get all resources directly. Note this is different from the opinion I left at `screen_manager.js`, which suggest you not access the LockScreen directly, because ScreenManager is totally a different component with different (but coherent) functions from LockScreen, but these sub-components are all related to LockScreen itself.